everware/home_handler.py

```diff
@@ -58,6 +61,27 @@ def get(self):
             branch_name = user.spawner.branch_name
             commit_sha = user.spawner.commit_sha
             repo_url = user.spawner.repo_url
+
+        if user.running and do_commit:
+            spawner = user.spawner
+            image_tag = datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+            image_name = 'everware_image/' + spawner.escaped_name + '/' + spawner.escaped_repo_url + '_' + spawner.container_id
+            host_with_protocol = self.request.protocol + '://' + self.request.host
+            url_with_image = url_concat(host_with_protocol + '/hub/spawn', dict(repourl='docker:' + image_name + ':' + image_tag))
+
+            self.log.info('Will commit %s' % url_with_image)
+
+            spawner.client.commit(
```

Comment:
why do you call it synchronously? it will make the whole service unresponsive for all users until it stops. consider ```yield user.spawner.docker``` (see github_agent or home_handler for example)